plugins/ec/README.md

```diff
@@ -0,0 +1,61 @@
+EC
+==
+
+This plugin provides flashing support for Embedded Controllers that use
+custom protocols.
+
+See https://en.wikipedia.org/wiki/Embedded_controller for more details about
+what the EC actually does.
+
+GUID Generation
+---------------
+
+These devices use a custom GUID generated using the EC chipset name:
+
+ * `EC-$(chipset)`, for example `EC-IT5570`
+
+Update Behavior
+---------------
+
+The firmware is deployed when the device is in normal runtime mode, but it is
+only activated on machine reboot. The firmware write is normally scheduled to be
+done very early in the boot process to minimize the chance the EC chip locking
+up if the user is actually using the keyboard controller.
+
+Vendor ID Security
+------------------
+
+The vendor ID is set from the baseboard vendor, for example `DMI:Star Labs`.
+
+Configuration
+-------------
+
+The following parameters can be specified in `/etc/fwupd/ec.conf`.
+
+**AutoloadAction** parameter can be set to `none`, `disable`, `seton` or
+`setoff`. `none` is the default.
+
+**DoNotRequireAC** parameter is boolean (`true` or `false`).  `false` is the
+default.
```

Comment:
can you explain these both a bit please? e.g. DoNotRequireAC is confusing as we already allow the user to --force the update on battery power.

Reply:
I don't really know anything about `AutoloadAction` except that we disable it according to update instructions.

`DoNotRequireAC` is for internal check which I added due to this error:
```
23:16:36:0205 FuPluginUpower       failed to query power type
```
So builtin check doesn't work in this case (`upowerd` is up and running).

Reply:
Okay, we don't want to workaround the upower failure in the plugin. I'll remove that.

Can you find out more what AutoloadAction is actually for? I don't know if it's supposed to be per-machine (i.e. a DMI match) or per-chipset, i.e. in the quirk file. I feel like we don't want a config entry for this.

Reply:
@hughsie Don't know much more what it does currently, but it looks to me like it's rather chipset-specific. And for the IT5570 should be set to disable according to the official update instructions.

Reply:
Quirk is rather a better way than config indeed.